docs/state.rst

```diff
@@ -244,3 +246,119 @@ For example:
         otherAppOtherAccountRole,
         If(otherAppOtherAccountRole.hasValue(), otherAppOtherAccountRole.value(), Bytes("none"))
     ])
+
+Box Storage
+-----------
+
+Box storage consists of key-value pairs that are stored in application's local context.
+
+The app account's minimum balance requirement (MBR) is increased by each box, and by the byte size of the box.
+If one deletes an application with outstanding boxes, the MBR is not recoverable from the deleted app account.
+
+Box sizes and names cannot be changed after initial allocation, but they can be deleted and re-allocated.
+Boxes are only visible to the app itself, other apps cannot view the app's own boxes on-chain.
+
+Boxes can be manipulated as follows:
+
+Create Boxes
+~~~~~~~~~~~~
+
+To create a box, use :any:`App.box_create`, or :any:`App.box_put` method.
+
+For :any:`App.box_create`, the first argument is the box name, and the second argument is the byte size to be allocated.
+
+For example:
+
+.. code-block:: python
+
+    # Allocate a box called "BoxForAlice" of byte size 100
+    App.box_create(Bytes("BoxForAlice"), Int(0xA11CE))
+    # Allocate a box called "BoxForBob" of byte size 90
+    App.box_create(Bytes("BoxForBob"), Int(0xB0B)
+
+For :any:`App.box_put`, the first argument is the box name to create or to write to, and the second argument is the bytes to write.
+
+**Notice**: if the box exists, then :any:`App.box_put` will write the contents to the box;
```

Comment:
```suggestion
**Notice**: if the box exists, then :any:`App.box_put` will write the contents to the box (assuming the replacement has length identical to the box'es size, as otherwise the operation will fail);
```